cohortextractor2/dsl.py

```diff
@@ -222,6 +214,18 @@ def __hash__(self) -> int:
         return hash(repr(self.value))
 
 
+class Predicate:
+    def __init__(
+        self, column: Column, operator: str, other: str | bool | Codelist | None
```

Comment:
Must the LHS of the predicate be a `Column`?

This means we can do `.filter(col > 123)` but can't do `.filter(123 < col)`.

Reply:
Also: we're going to want to use `PatientSeries` in predicates.

For instance, in the example I documented we've got `covid_imms.filter(imms.date > (cohort.first_date + 28))`.  `imms.date` is a `Column`, but `cohort.first_date` and `cohort.first_date + 28` are `PatientSeries`.

Reply:
It seems that writing the comparison the other way round just works (TM).

Reply:
I've added a separate story for using patient series in predicates.